File: C2109L/IDP/de01/employee.dart
import 'dart:io';

class Employee {
  String id;
  String name;
  int workingDays;
  double dailySalary;
  Employee({
    required this.id,
    required this.name,
    required this.workingDays,
    required this.dailySalary
  });
  factory Employee.input() {
    String _id = "";
    while(_id.length != 6) {
      print('Enter id: ');
      _id = stdin.readLineSync() ?? '';
    }
    String _name = "";
    while(_name.length < 5 || _name.length > 100) {
      print('Enter name: ');
      _name = stdin.readLineSync() ?? '';
    }
    int _workingDays = 0;
    while(_workingDays < 1 || _workingDays > 31) {
      print('Enter _workingDays: ');
      _workingDays = int.parse(stdin.readLineSync() ?? '0');
    }
    double _dailySalary = 0;
    while(_dailySalary < 10 || _dailySalary > 100) {
      print('Enter _workingDays: ');
      _dailySalary = double.parse(stdin.readLineSync() ?? '0');
    }
    return Employee(
        id: _id,
        name: _name,
        workingDays: _workingDays,
        dailySalary: _dailySalary
    );
  }
  @override
  String toString() {
    // TODO: implement toString
    return 'id: $id, name: $name, workingDays: $workingDays, dailySalary: $dailySalary';
  }
  //monthly_salary = (workingDays x dailySalary)
  double get monthlySalary => workingDays * dailySalary;
}

File: C1908GLeThanhNghi/AP/18-11-2020/ConsoleApp/ConsoleApp/Employee.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ConsoleApp
{
    public class Employee
    {
        private string name;
        public int Id { get; set; } //Properties
        //full getter/setter
        /*
        public string Name { 
            get {
                Console.WriteLine("This is getter");
                return name;
            } 
            set {
                Console.WriteLine("This is setter");
                name = value.ToUpper();
            } 
        }
        */
        public string Name { get; set; }
        public int WorkingDays { get; set; }
        public float DailySalary { get; set; }
        //factory method
        public static Employee InputEmployee()
        {
            Console.WriteLine("Enter id : ");
            int id = Convert.ToInt32(Console.ReadLine().Trim());
            Console.WriteLine("Enter name : ");
            string name = Console.ReadLine().Trim();
            Console.WriteLine("Enter workingDays : ");
            int workingDays = Convert.ToInt32(Console.ReadLine());
            Console.WriteLine("Enter dailySalary : ");
            float dailySalary = (float)Convert.ToDouble(Console.ReadLine());
            return new Employee()
            {
                Id = id, Name = name, 
                WorkingDays = workingDays, DailySalary = dailySalary
            };

        }

        public override string ToString()
        {
            return $"Name = {Name}, " +
                $"Id = {Id}, " +
                $"WorkingDays = {WorkingDays}, " +
                $"DailySalary = {DailySalary}";
        }
    }
}
